### PR TITLE
fix: TextShape outline not working

### LINKED ISSUE #1908: [QA] TextShape Outline Not Working - Validate Properties
### **Explorer Alpha build versions:**

- **v0.21.0**

### **Issue Description:**
In the old client, a TextShape with an outline renders properly. However, in the Explorer Alpha (EA), the outline does not appear. Each of the TextShape properties, including outlineWidth, outlineColor, fontSize, color, and others, should be validated to ensure they work as expected in EA.

### **STR:** 

1. Launch the Explorer Alpha.
2. Load a scene that includes a `TextShape` with an outline.
3. Observe the `TextShape` for the missing outline.
4. Check the behavior of other `TextShape` properties like `fontSize`, `color`, `shadowBlur`, etc.

### ❌ Actual Result

- The `TextShape` appears without the outline.
- Potential issues with other `TextShape` properties in Explorer Alpha.

image (2)

### ✅ Expected Result

- The `TextShape` should display with its outline as configured.
- All `TextShape` properties should behave consistently, matching the old client.

image (1)

### PR DESCRIPTION
**Pull request by sandrade-dcl**
## What does this PR change?
Fix #1908 

In the Explorer Alpha, the TextShape outline did not appear.
1. Launch the Explorer Alpha.
2. Load a scene that includes a TextShape with an outline (for example `-132,-42`).
3. Check that you see the outline correctly.

## Our Code Review Standards
https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

## Summary by CodeRabbit

- **New Features**
	- Enhanced control over text shape rendering with conditional outline and underlay effects based on specific properties.
	- Improved performance and clarity in text shape rendering logic.

- **Bug Fixes**
	- Resolved issues with unnecessary rendering of effects when properties are set to zero.

### CODE REVIEW

File: Explorer/Assets/DCL/SDKComponents/TextShape/TMPProSdkExtensions.cs [hunk 1]
```diff
@@ -13,8 +13,8 @@ public static class TMPProSdkExtensions
         private static readonly int ID_OUTLINE_WIDTH = Shader.PropertyToID("_OutlineWidth");
         private static readonly int ID_UNDERLAY_COLOR = Shader.PropertyToID("_UnderlayColor");
         private static readonly int ID_UNDERLAY_SOFTNESS = Shader.PropertyToID("_UnderlaySoftness");
-        private static readonly int ID_UNDERLAY_OFFSET_Y = Shader.PropertyToID("_UnderlayOffsetY");
-        private static readonly int ID_UNDERLAY_OFFSET_X = Shader.PropertyToID("_UnderlayOffsetX");
```
Comment (AlejandroAlvarezMelucciDCL) [inline Explorer/Assets/DCL/SDKComponents/TextShape/TMPProSdkExtensions.cs:17]: Are we sure we want to remove this?

File: Explorer/Assets/DCL/SDKComponents/TextShape/TMPProSdkExtensions.cs [hunk 2]
```diff
@@ -54,25 +54,27 @@ public static void Apply(this TextMeshPro tmpText, PBTextShape textShape, IFonts
...
-            // TODO (Vit) : Disable outline when if (textShape.OutlineWidth <= 0)
-            materialPropertyBlock.SetColor(ID_OUTLINE_COLOR, textShape.OutlineColor?.ToUnityColor() ?? Color.white);
-            materialPropertyBlock.SetFloat(ID_OUTLINE_WIDTH, textShape.OutlineWidth);
+            tmpText.renderer.GetPropertyBlock(materialPropertyBlock);
 
-            // TODO (Vit ): disable shadow when if (textShape.ShadowOffsetX == 0 || textShape.ShadowOffsetY == 0)
-            materialPropertyBlock.SetColor(ID_UNDERLAY_COLOR, textShape.ShadowColor?.ToUnityColor() ?? Color.white);
-            materialPropertyBlock.SetFloat(ID_UNDERLAY_SOFTNESS, textShape.ShadowBlur);
-            materialPropertyBlock.SetFloat(ID_UNDERLAY_OFFSET_X, textShape.ShadowOffsetX);
-            materialPropertyBlock.SetFloat(ID_UNDERLAY_OFFSET_Y, textShape.ShadowOffsetY);
```
Comment (AlejandroAlvarezMelucciDCL) [inline Explorer/Assets/DCL/SDKComponents/TextShape/TMPProSdkExtensions.cs:73]: We're missing the X and Y shadow offset, is there a reason to remove it?

**Review by AlejandroAlvarezMelucciDCL — APPROVED** [review]
LGTM but left some questions about `shadow offset`

File: Explorer/Assets/DCL/SDKComponents/TextShape/TMPProSdkExtensions.cs (hunk 2 above)
Comment (sandrade-dcl) [reply Explorer/Assets/DCL/SDKComponents/TextShape/TMPProSdkExtensions.cs:73]: You're are completely right! Trying things and different aaproaches in the code I forgot to keep these 2 properties. I've added them back again: https://github.com/decentraland/unity-explorer/pull/2007/commits/783ee9a1fecba5f64c505ea692826cb411701cf9

**Pull request merged**